pxr/usd/usdSkel/validators.cpp

```diff
@@ -0,0 +1,91 @@
+//
+// Copyright 2024 Pixar
+//
+// Licensed under the terms set forth in the LICENSE.txt file available at
+// https://openusd.org/license.
+//
+
+#include "pxr/usd/sdr/shaderProperty.h"
+#include "pxr/usd/usd/prim.h"
+#include "pxr/usd/usd/schemaRegistry.h"
+#include "pxr/usd/usd/validationError.h"
+#include "pxr/usd/usd/validationRegistry.h"
+#include "pxr/usd/usd/validator.h"
+#include "pxr/usd/sdr/registry.h"
+#include "pxr/base/tf/token.h"
+#include "pxr/base/tf/stringUtils.h"
+#include "pxr/usd/usdSkel/validatorTokens.h"
+#include "pxr/usd/usdSkel/bindingAPI.h"
+
+#include <algorithm>
+#include <unordered_map>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+static
+UsdValidationErrorVector
+_SkelBindingApiChecker(const UsdPrim &usdPrim)
+{
+    UsdValidationErrorVector errors;
+
+    if (!usdPrim.HasAPI<UsdSkelBindingAPI>()){
+        UsdSchemaRegistry& usdSchemaRegistry = UsdSchemaRegistry::GetInstance();
+        std::unique_ptr<UsdPrimDefinition> primDef = usdSchemaRegistry.BuildComposedPrimDefinition(TfToken(), {TfToken("SkelBindingAPI")});
+
+        std::vector<TfToken> skelPropertyNames = primDef->GetPropertyNames();
+        std::vector<TfToken> primPropertyNames = usdPrim.GetPropertyNames();
+
+        for (const TfToken &primToken : primPropertyNames){
+            for (const TfToken &skelToken : skelPropertyNames){
+                if (skelToken != primToken)
+                    continue;
+
+                errors.emplace_back(
+                        UsdValidationErrorType::Error,
+                        UsdValidationErrorSites{
+                                UsdValidationErrorSite(usdPrim.GetStage(),
+                                                       usdPrim.GetPath())
+                        },
+                        TfStringPrintf(("Found a UsdSkelBinding property (%s), "
+                            "but no SkelBindingAPI applied on the prim <%s>.(fails 'SkelBindingAPIAppliedChecker')"),
```

Comment:
I will also recommend to remove the test name from the error string. Validator which resulted in this error will automatically be embedded in the Error (https://github.com/PixarAnimationStudios/OpenUSD/blob/dev/pxr/usd/usd/validator.cpp#L61).

Reply:
removed